mandatory/3-playing-computer.js

```diff
@@ -7,36 +7,47 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
-  2. Remove the line that throws the error.
+    [ console.log(b) throws error as b is undefined]
+  2. Remove the line that throws the error. 
+    [commented out the line causing errors]
   3. What is printed to the console?
+    [2, 6, 6, 4, 7, 9, 8, 6, 9, 13, 10, 8]
   4. How many times is "f1" called?
+    [10 times, used console.log("find", a)]
   5. How many times is "f2" called?
+    [2 times, used console.log("search", x)]
```

Comment:
It's called 3 times. As mentioned above the conditional checks if the index is odd or even. The index is even 3 times (0,2,4) and so f2 is called 3 times.